nixos/modules/services/mail/davmail.nix

```diff
@@ -0,0 +1,36 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.davmail;
+in
+  {
+    options.services.davmail = {
+      enable = mkEnableOption ''
+        DavMail is a POP/IMAP/SMTP/Caldav/Carddav/LDAP exchange gateway
+        allowing users to use any mail/calendar client with an Exchange server.
+      '';
+
+      configFile = mkOption {
+        default = null;
+        type = with types; nullOr path;
+        description = "Path to the davmail configuration file.";
+      };
+    };
+
+    config = mkIf cfg.enable {
+      systemd.services.davmail = {
+        description = "DavMail POP/IMAP/SMTP Exchange Gateway";
+        serviceConfig = {
+          Type = "simple";
+          ExecStart = "${pkgs.davmail}/bin/davmail"
+            + optionalString (cfg.configFile != null) " ${cfg.configFile}";
+          Restart = "on-failure";
```

Comment:
Without specifying a user, this service will run as root. Try using `DynamicUser` to prevent that (and possibly `StateDirectory` and similar).

Reply:
Or alternatively, creating a new user as many services do too (but if DynamicUser would work that would be awesome)

Reply:
I just pushed changes to address your other comments. I'll test out `DynamicUser` when I get a chance.